fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/observer/DefaultFilter.java

```diff
@@ -95,9 +91,13 @@ public boolean apply(final Event event) {
         }
     }
 
-    private static org.modeshape.jcr.api.observation.Event getJcr21Event(final Event event) {
+    protected static Collection<String> getMixinTypes(final Event event)
+            throws PathNotFoundException, RepositoryException {
         try {
-            return (org.modeshape.jcr.api.observation.Event) event;
+            final org.modeshape.jcr.api.observation.Event modeEvent =
+                    (org.modeshape.jcr.api.observation.Event) event;
+            return ImmutableSet.copyOf(transform(ImmutableList.copyOf(modeEvent.getMixinNodeTypes()),
```

Comment:
That's what the code was doing before -- but we could just remove that extra step and return the List.